From 770a434613caa7073316e249a4c76bfb974df249 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mika=20Bostr=C3=B6m?= <bostik@odin.bostik.iki.fi>
Date: Thu, 27 Dec 2012 09:34:37 +0200
Subject: [PATCH 8/9] Add build options to WebKit.pro

Some of the larger features are controlled by WEBKIT_CONFIG. We set them
explicitly to (hopefully) reasonable values.
---
 WebKit.pro |   19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/WebKit.pro b/WebKit.pro
index e81dcfe..5f95d46 100644
--- a/WebKit.pro
+++ b/WebKit.pro
@@ -5,9 +5,28 @@
 # See 'Tools/qmake/README' for an overview of the build system
 # -------------------------------------------------------------------
 
+# QtWebKit build options
+# build?(feature) checks for "build_feature" in WEBKIT_CONFIG
+# have?(feature) checks for "have_feature" in WEBKIT_CONFIG
+# use?(feature) checks for "use_feature" in WEBKIT_CONFIG
+# enable?(feature) checks for "feature" in WEBKIT_CONFIG
+#                            ^^^^^^^^^^^
+# NOTE: there is no prefix in for enabled features
+#
+WEBKIT_CONFIG += fullscreen_api
+WEBKIT_CONFIG += build_webkit2
+WEBKIT_CONFIG -= netscape_plugin_api
+WEBKIT_CONFIG += use_3d_graphics
+
+
 # Try to reduce linking memory requirements even further
 CONFIG += force_static_libs_as_shared
 
+# Build for release only
+CONFIG -= debug
+CONFIG -= debug_and_release
+CONFIG += release
+
 TEMPLATE = subdirs
 CONFIG += ordered
 
-- 
1.7.10.4

